rust/connection-tests/src/main.rs

```diff
@@ -0,0 +1,389 @@
+use std::{
+    future::poll_fn,
+    net::{IpAddr, Ipv4Addr},
+    str::FromStr,
+    task::{Context, Poll},
+    time::{Duration, Instant},
+};
+
+use anyhow::{bail, Context as _, Result};
+use boringtun::x25519::{PublicKey, StaticSecret};
+use firezone_connection::{
+    Answer, ClientConnectionPool, ConnectionPool, Credentials, IpPacket, Offer,
+    ServerConnectionPool,
+};
+use futures::{future::BoxFuture, FutureExt};
+use pnet_packet::{ip::IpNextHeaderProtocols, ipv4::Ipv4Packet};
+use redis::AsyncCommands;
+use secrecy::{ExposeSecret as _, Secret};
+use tokio::{io::ReadBuf, net::UdpSocket};
+use tracing_subscriber::EnvFilter;
+
+const MAX_UDP_SIZE: usize = (1 << 16) - 1;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    tokio::time::sleep(Duration::from_secs(1)).await; // Until redis is up.
```

Comment:
Can we depend on a healthcheck instead?

Reply:
I tried that but for some reason, couldn't get it to work. If you don't mind, I'll merge this as is and have a play with it later!